20200928 Fixes - 40mms, Vendors, Weapon Fixes

5crownik007 opened the pull request:
Closes #2815 
🆑 5crownik007
tweak: 40mm available for purchase in cargo shuttle. landmines are available too
tweak: 40mm illumination shells now illuminate point of impact
tweak: ODST vendors now vend M6S mags instead of M6D mags
bugfix: body scanners now function
tweak: covenant now have ample supply of armourkits
bugfix: ma3 fixed finally
tweak: unsc plasma weapons have damage parity with their ballistic counterparts
/🆑

5crownik007 committed:
bodyscan console fix

bodyscanner consoles now search the dir they are facing for their connected bodyscanner

5crownik007 committed:
odst m6s ammo

odstvendor now vends correct magazines

5crownik007 committed:
40mm supplypacks

40mm he now available in cargo

5crownik007 committed:
supplypack boxes

5crownik007 committed:
MA3 actually works now

numerous code fixes, appears to work as intended now

5crownik007 committed:
ma37 fixes

magazines use proper sprites, and have correct capacity

5crownik007 committed:
addresses #2815

armourkits are now numerous

5crownik007 committed:
40mm illum functionality

gives 40mm illumination shells functionality in the form of a flare

5crownik007 committed:
plasma weapon parity

unsc plasma weapons now have parity with their ballistic counterparts in terms of damage

X0-11 commented on code/modules/halo/weapons/ammo_g40mm.dm:
```diff
@@ -36,6 +36,15 @@
+	light_color = "#ffff99"
+	icon_state = "40mm_shell_illumination"
+	item_state = null
+	on = 1
```
better to make the code that spawns this flare activate it normally

5crownik007 replied:
why? This works just fine from testing

5crownik007 replied:
Eh there are probably unintended consequences I didn't account for you're right

X0-11 replied:
flarecode activation may have special handling effects of it's own or have them added later. Utilising the generalised method of activation would be better

5crownik007 replied:
done

5crownik007 committed:
40mm flarecode

40mm flares now use existing turn_on() proc to activate once created
They do indeed run out of fuel, though I don't know if they didn't do that before

5crownik007 committed:
landmines

Landmines are available in the supply shuttle

bloxgate commented on code/modules/halo/unsc/supply/ammo.dm:
```diff
@@ -141,14 +141,40 @@
+/* M301 */
+
+/decl/hierarchy/supply_pack/unsc_ammo/g40mm_he
+	name = "M301 40mm HE crate(1 box)"
```
Personally, I'd add a space before the `(1 box)`

5crownik007 replied:
done

bloxgate commented on code/modules/halo/weapons/unsc_plasma.dm:
```diff
@@ -133,7 +133,8 @@
 	icon_state = "plasma_proj_light"
-	damage = 25
+	damage = 30
+	shield_damage = -5
```
Is this supposed to be negative?

5crownik007 replied:
Yes

bloxgate commented on code/modules/halo/covenant/structures_machines/covvendors.dm:
```diff
@@ -145,8 +145,6 @@
 		/obj/item/dumb_ai_chip/cov = 2,
 		/obj/item/weapon/plastique/covenant = 8,
-		/obj/item/weapon/armor_patch/cov = 8,
-		/obj/item/weapon/armor_patch/mini/cov = 10,
```
Unless I'm missing something about how this works, doesn't this mean the covenant now have no armor kits, with the exception of in the Kig-Yar vendors?

I was under the impression that the issue was them having too few, not too many.

5crownik007 replied:
pointbased vendors default to 100 in a pool iirc if amount is not set, you are missing something

5crownik007 committed:
formatting
